Mathlib/Analysis/Complex/Hadamard.lean

```diff
@@ -0,0 +1,488 @@
+/-
+Copyright (c) 2023 Xavier Généreux. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Xavier Généreux
+-/
+
+import Mathlib.Analysis.SpecialFunctions.Pow.Deriv
+import Mathlib.Analysis.Complex.PhragmenLindelof
+
+/-!
+# Hadamard three-lines Theorem
+
+In this file we present a proof of a special case Hadamard's three-lines Theorem.
+
+## Main result
+
+- `abs_le_interp_on_closedStrip` :
+Hadamard three-line theorem on `re ⁻¹' [0,1]`: If `f` is a bounded function, continuous on
+`re ⁻¹' [0,1]` and differentiable on `re ⁻¹' (0,1)`, then for
+`M(x) := sup ((norm ∘ f) '' (re ⁻¹' {x}))`, that is `M(x)` is the supremum of the absolute value of
+`f` along the vertical lines `re z = x`, we have that `∀ z ∈ re ⁻¹' [0,1]` the inequality
+`|f(z)| ≤ |M(0)^(1-z)| * |M(1)^z|` holds. This can be seen to be equivalent to the statement
+that `log M(re z)` is a convex function on `[0,1]`.
+
+## Main definitions
+
+- `Complex.HadamardThreeLines.strip` : The vertical strip defined by : `re ⁻¹' Ioo a b`
+
+- `Complex.HadamardThreeLines.closedStrip` : The vertical strip defined by : `re ⁻¹' Icc a b`
+
+- `Complex.HadamardThreeLines.sSupAbsIm` :
+    The supremum function on vertical lines defined by : `sSup {|f(z)| : z.re = x}`
+
+- `Complex.HadamardThreeLines.invInterpStrip` :
+    Inverse of the interpolation between the `sSupAbsIm` on the edges of the strip.
+
+- `Complex.HadamardThreeLines.F` :
+    Function defined by `f` times `invInterpStrip`. Convenient form for proofs.
+
+- `Complex.HadamardThreeLines.interpStrip` :
+    The interpolation between the `sSupAbsIm` on the edges of the strip.
+
+- `Complex.HadamardThreeLines.transl`
+    A function translated by a constant `ε`.
+
+## References
+
+The proof follows from Phragmén-Lindelöf when both frontiers are not everywhere zero.
+We then use a limit argument to cover the case when either of the sides are `0`.
+-/
+
+
+open Set Filter Function Complex
+
+namespace Complex
+namespace HadamardThreeLines
+variable (z : ℂ)
+
+/-- The strip in the complex plane containing all `z ∈ ℂ` such that `z.re ∈ Ioo a b`. -/
+def strip (a : ℝ) (b : ℝ) : Set ℂ := re ⁻¹' Ioo a b
```

Comment:
```suggestion
def verticalStrip (a : ℝ) (b : ℝ) : Set ℂ := re ⁻¹' Ioo a b
```
I can imagine that some day people will also want to use horizontal strips, so let's lift the ambiguity from the start. Same thing everywhere below.